onadata/libs/tests/utils/test_export_builder.py

```diff
@@ -527,6 +527,59 @@ def test_zipped_sav_export_with_date_field(self):
 
         shutil.rmtree(temp_dir)
 
+    def test_zipped_sav_export_dynamic_select_multiple(self):
+        md = """
+        | survey |
+        |        | type                  | name | label                             | calculation                       |
+        |        | select_one sex        | sex | Are you male or female?           |                                   |
+        |        | calculate             | text |                                   | if((${sex}=’male’), 'his', ‘her’) |
+        |        | text                  | name | What's ${text} name?              |                                   |
+        |        | select_multiple brand | brand_known | Brands known in category   |                                   |
+
+        | choices |
+        |         | list name | name   | label  |
+        |         | sex       | male   | Male   |
+        |         | sex       | female | Female |
+        |         | brand     | ${text}| ${text}|
+        |         | brand     | a      | a      |
+        |         | brand     | b      | b      |
+        """
+        survey = self.md_to_pyxform_survey(md, {'name': 'exp'})
+        data = [{"sex": "male", "name": "Davis", "brand_known": "his a"}]
+        export_builder = ExportBuilder()
+        export_builder.set_survey(survey)
+        temp_zip_file = NamedTemporaryFile(suffix='.zip')
+        export_builder.to_zipped_sav(temp_zip_file.name, data)
+        temp_zip_file.seek(0)
+        temp_dir = tempfile.mkdtemp()
+        zip_file = zipfile.ZipFile(temp_zip_file.name, "r")
+        zip_file.extractall(temp_dir)
+        zip_file.close()
+        temp_zip_file.close()
+
+        self.assertTrue(
+            os.path.exists(
+                os.path.join(temp_dir, "exp.sav")))
+
+        with SavReader(os.path.join(temp_dir, "exp.sav"),
+                       returnHeader=True) as reader:
+            rows = [r for r in reader]
+            self.assertTrue(len(rows) > 1)
+            self.assertEqual(rows[0][0], b'sex')
+            self.assertEqual(rows[1][0], b'male')
+            self.assertEqual(rows[0][2], b'name')
+            self.assertEqual(rows[1][2], b'Davis')
+            self.assertEqual(rows[0][3], b'brand_known')
+            self.assertEqual(rows[1][3], b'his a')
+            self.assertEqual(rows[0][4], b'brand_known.his')
```

Comment:
Challenge: The choice is dynamic... The choice could be `his` or `her` and while generating the export currently this field is set to `brand_known.${text}` the ideal situation would be to link the dynamic variable `text` somehow.